From 54ab4796f68bda4e894ce9aef2adbd2b28508a0e Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:37 -0400
Subject: [ethernet] cxgb4: remove unnecessary null test before debugfs_remove_recursive

Message-id: <1410475447-94392-79-git-send-email-svemuri@redhat.com>
Patchwork-id: 93840
O-Subject: [RHEL7.1 PATCH BZ 1124947 078/108] drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c: remove unnecessary null test before debugfs_remove_recursive
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Fix checkpatch warning:
"WARNING: debugfs_remove_recursive(NULL) is safe this check is probably not required"

Cc: Hariprasad S <hariprasad@chelsio.com>
Cc: netdev@vger.kernel.org
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 9f16dc2ec7cb3527c66581ad762876ba1f774cdb)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 7eadcbc..438abc1 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -6441,8 +6441,7 @@ static void remove_one(struct pci_dev *pdev)
    if (adapter->port[i]->reg_state == NETREG_REGISTERED)
     unregister_netdev(adapter->port[i]);
 
-  if (adapter->debugfs_root)
-   debugfs_remove_recursive(adapter->debugfs_root);
+  debugfs_remove_recursive(adapter->debugfs_root);
 
   /* If we allocated filters, free up state associated with any
    * valid filters ...
-- 
1.7.1